src/backend/common/util.cpp

```diff
@@ -91,3 +102,114 @@ std::string int_version_to_string(int version) {
     return std::to_string(version / 1000) + "." +
            std::to_string((int)((version % 1000) / 10.));
 }
+
+#if defined(OS_WIN)
+string getTemporaryDirectory() {
+    DWORD bufSize = 261;  // limit according to GetTempPath documentation
+    string retVal;
+    retVal.resize(bufSize);
+    bufSize = GetTempPathA(bufSize, &retVal[0]);
+    retVal.resize(bufSize);
+    return retVal;
+}
+#else
+string getHomeDirectory() {
+    string home = getEnvVar("XDG_CACHE_HOME");
+    if (!home.empty()) return home;
+
+    home = getEnvVar("HOME");
+    if (!home.empty()) return home;
+
+    home = getpwuid(getuid())->pw_dir;
```

Comment:
This value is not being returned. I was getting a warning when building on linux.

Reply:
Yeah, missing return for third case.

Reply:
Fixed